presto-main/src/test/java/io/prestosql/sql/planner/TestLogicalPlanner.java

```diff
@@ -801,4 +805,71 @@ public void testDistributedSort()
                                                 tableScan("orders", ImmutableMap.of(
                                                         "ORDERKEY", "orderkey")))))));
     }
+
+    @Test
+    public void testRemoveRedundantLimitNode()
+    {
+        assertFalse(
+                searchFrom(plan("SELECT count(*) FROM orders LIMIT 10", LogicalPlanner.Stage.OPTIMIZED).getRoot())
+                        .where(isInstanceOfAny(LimitNode.class))
+                        .matches(),
+                "Unexpected node for the above query");
+
+        assertPlan(
+                "SELECT orderkey, count(*) FROM orders GROUP BY orderkey LIMIT 10",
+                output(
+                        limit(10,
+                                anyTree(
+                                        tableScan("orders")))));
+    }
+
+    @Test
+    public void testRemoveSingleRowSort()
+    {
+        assertFalse(
+                searchFrom(plan("SELECT count(*) FROM orders ORDER BY 1", LogicalPlanner.Stage.OPTIMIZED).getRoot())
+                        .where(isInstanceOfAny(SortNode.class))
+                        .matches(),
+                "Unexpected node for the above query");
```

Comment:
`Unexpected node for the above query` -> `format("Unexpected sort node for query: '%s'", query)`. To the same for all below and above.